service/operator/service.go

```diff
@@ -387,23 +402,25 @@ func (s *Service) GetDeleteState(obj, currentState, desiredState interface{}) (i
 		{
 			newData := map[string]string{}
 			for k, v := range deleteState.ConfigMap.Data {
-				if inConfigMapData(dState.ConfigMapData, k, v) {
+				if !inConfigMapData(dState.ConfigMapData, k, v) {
```

Comment:
This would indeed not make sense. In case of the Flannel operator `GetDeleteState` would contain the items to be deleted. `ProcessDeleteState` would then iterate over the items and just remove them. The reason why we have to prepare the whole set here is that we have to update data of an existing resource.